core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.domains { passwordValidityPeriodInDays == 2147483647 }
+          ```
+      remediation: |
+       ### MS365 Console
+
+        To update via the ms365 portal:
+
+        1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+        2. Settings --> Org Settings --> Security & Privacy --> Password expiration policy -->  Set passwords to never expire (recommended)
+        3. Click Save 
+    query: |
+      msgraph.beta.domains {
+        passwordValidityPeriodInDays == 2147483647
+      }
+  - uid: mondoo-m365-ensure-multifactor-authentication-is-enabled-for-all-users-in-administrative-roles
+    title: Ensure all users in administrative roles have MFA enabled 
+    severity: 100
+    docs:
+      desc: |
+        Enable multifactor authentication for all users in administrative roles!
+      audit: |
+       __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10)" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10 )
+          ```
+      remediation: |
+         ### MS365 Console
+         
+         To update via the ms365 portal:
+
+         1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+         2. Click on "Azure Active Directory" once in the "Admin Centers"
+         3. Applications --> Enterprise applications --> Security --> Conditional Access 
+         4. Make sure there is a Policy with following conditions: 
+         
+           * Grant --> Grant access (Require multifactor authentication)
+           * Users --> Users and groups --> Include --> Select users and groups --> Directory roles --> at minimum select following rules (Billing admin, Conditional Access admin, Exchange admin, Global admin, Helpdesk admin, Security admin, SharePoint admin, and User admin)
+           * Cloud apps or actions --> All cloud apps 
+         5. Make sure that the policy is Enabled  
+    query: |
+      msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10 )
```

Comment:
`msgraph.beta.security` is deprecated and should be replaced with the latest version

Reply:
Thanks Chris, done!